### PR TITLE
Files sharing event dispatcher

### PR DESCRIPTION
**Pull request by rullzer**
As requested already some time ago by @oparoz 

CC: @LukasReschke @nickvergessen @MorrisJobke

### CODE REVIEW

**Comment by rullzer** [comment]
Aaah the getmock stuff is probaly conflicting with https://github.com/nextcloud/server/pull/409 but lets see when that is merged.

File: apps/files_sharing/lib/Controllers/ShareController.php [hunk 1]
```diff
@@ -108,7 +111,8 @@ public function __construct($appName,
 								ISession $session,
 								IPreview $previewManager,
 								IRootFolder $rootFolder,
-								FederatedShareProvider $federatedShareProvider) {
+								FederatedShareProvider $federatedShareProvider,
+								EventDispatcherInterface $eventDispatcher) {
```
Comment (LukasReschke) [inline apps/files_sharing/lib/Controllers/ShareController.php]: PHPDoc missing 😉

Comment (rullzer) [inline apps/files_sharing/lib/Controllers/ShareController.php]: fixed

File: apps/files_sharing/lib/Controllers/ShareController.php [hunk 2]
```diff
@@ -353,6 +359,31 @@ public function showShare($token, $path = '') {
 		$shareTmpl['previewMaxX'] = $this->config->getSystemValue('preview_max_x', 1024);
 		$shareTmpl['previewMaxY'] = $this->config->getSystemValue('preview_max_y', 1024);
 
+		// Load files we need
+		OCP\Util::addScript('files', 'file-upload');
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controllers/ShareController.php]: Please kill `use OC;` and `use OCP;`
Ridiculous stuff...

**Comment by MorrisJobke** [comment]
Tested and still works 👍

File: apps/files_sharing/lib/Controllers/ShareController.php (hunk 2 above)
Comment (rullzer) [inline apps/files_sharing/lib/Controllers/ShareController.php]: ?? It is OCP here...

Stuff like this sould also be handled via DI from my POV btw...

Comment (nickvergessen) [inline apps/files_sharing/lib/Controllers/ShareController.php]: I mean the use statement, either you write `\OCP\Util` or `Util`, but not `OCP\Util`+ `use OCP;`

Comment (rullzer) [inline apps/files_sharing/lib/Controllers/ShareController.php]: Aaah ok. Got it. Fixed. And cleaned up some other stuff as well.

**Comment by rullzer** [comment]
Rebased to make merging possible ;)

**Comment by schiessle** [comment]
looks good :+1:

**Pull request merged**